src/index.ts

```diff
@@ -1,228 +1,7 @@
-   */
-  async callback(
-    code: string,
-    nonce: string | null = null,
-    redirectUri = this.getFirstRedirectUri(),
-  ): Promise<{ sub: string; accessToken: string }> {
-    const tokenSet = await this.sgID.callback(
-      redirectUri,
-      { code },
-      { nonce: nonce ?? undefined },
-    )
-    const { sub } = tokenSet.claims()
-    const { access_token: accessToken } = tokenSet
-    if (!sub) {
-      throw new Error(Errors.NO_SUB_ERROR)
-    }
-    if (!accessToken) {
-      throw new Error(Errors.NO_ACCESS_TOKEN_ERROR)
-    }
-    return { sub, accessToken }
-  }
-
-  /**
-   * Retrieves verified user info and decrypts it with your private key.
-   * @param accessToken The access token returned in the callback function
-   * @returns The sub of the end-user and the end-user's verified data
-   */
-  async userinfo(
-    accessToken: string,
-  ): Promise<{ sub: string; data: Record<string, string> }> {
-    /**
-     * sub: user sub (also returned previously in id_token)
-     * encryptedPayloadKey: key encrypted with client's public key (for decrypting userinfo jwe)
-     * data: jwe of userinfo
-     */
-    const {
-      sub,
-      key: encryptedPayloadKey,
-      data,
-    } = await this.sgID.userinfo<{
-      sub: string | undefined
-      key: string | undefined
-      data: Record<string, string> | undefined
-    }>(accessToken)
-
-    if (encryptedPayloadKey && data) {
-      const result = await this.decryptPayload(encryptedPayloadKey, data)
-      return { sub, data: result }
-    }
-
-    return { sub, data: {} }
-  }
-
-  private async decryptPayload(
-    encryptedPayloadKey: string,
-    data: Record<string, string>,
-  ): Promise<Record<string, string>> {
-    let privateKeyJwk
-    let payloadJwk
-    try {
-      // Import client private key in PKCS8 format
-      privateKeyJwk = await importPKCS8(this.privateKey, 'RSA-OAEP-256')
-    } catch (e) {
-      throw new Error(Errors.PRIVATE_KEY_IMPORT_ERROR)
-    }
-
-    // Decrypt key to get plaintext symmetric key
-    const decoder = new TextDecoder()
-    try {
-      const decryptedKey = decoder.decode(
-        (await compactDecrypt(encryptedPayloadKey, privateKeyJwk)).plaintext,
-      )
-      payloadJwk = await importJWK(JSON.parse(decryptedKey))
-    } catch (e) {
-      throw new Error(Errors.DECRYPT_BLOCK_KEY_ERROR)
-    }
-
-    // Decrypt each jwe in body
-    const result: Record<string, string> = {}
-    try {
-      for (const field in data) {
-        const jwe = data[field]
-        const decryptedValue = decoder.decode(
-          (await compactDecrypt(jwe, payloadJwk)).plaintext,
-        )
-        result[field] = decryptedValue
-      }
-    } catch (e) {
-      throw new Error(Errors.DECRYPT_PAYLOAD_ERROR)
-    }
-    return result
-  }
-}
+import { SgidClient } from './sgidClient'
 
 export default SgidClient
+export * from './sgidClient'
+export * from './types'
+export * from './util'
+export * from './generators'
```

Comment:
since we are changing the API of the SDK, can we update:
1. The repo README
2. The PR description on why we decided to do so